tracecat/tables/service.py

```diff
@@ -370,6 +425,54 @@ async def update_column(
         await self.session.flush()
         return column
 
+    @require_access_level(AccessLevel.ADMIN)
+    async def create_unique_index(self, table: Table, column_names: list[str]) -> None:
+        """Create a unique index on specified columns."""
+        # Get the fully qualified table name with schema
+        full_table_name = self._full_table_name(table.name)
+
+        # Sanitize column names to prevent SQL injection
+        sanitized_columns = [self._sanitize_identifier(col) for col in column_names]
+
+        # Create a descriptive name for the index
+        # Format: uq_[table_name]_[col1]_[col2]_etc
+        if len(sanitized_columns) == 1:
+            # This is the pattern expected by _enrich_columns_with_natural_key_info
+            index_name = f"uq_{table.name}_{sanitized_columns[0]}"
+        else:
+            # Use a different prefix for multi-column indexes
+            index_name = "multiuq_{}_{}".format(table.name, "_".join(sanitized_columns))
+
+        # Get database connection
+        conn = await self.session.connection()
+
+        # Execute the CREATE UNIQUE INDEX SQL command
+        await conn.execute(
+            sa.DDL(
+                "CREATE UNIQUE INDEX %s ON %s (%s)",
+                (
+                    index_name,  # Name of the index
+                    full_table_name,  # Table to create index on
+                    ", ".join(sanitized_columns),  # Column(s) to index
+                ),
+            )
+        )
+
+        # Commit the transaction
+        await self.session.commit()
```

Comment:
can we flush here instead? I'd prefer is the BaseTablesService is non-transactional and only TablesService implements transactional versions.